### PR TITLE
[FLINK-5166] TextInputFormatTest.testNestedFileRead

### PR DESCRIPTION
**Pull request by shijinkui**
- [x] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [x] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [x] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

**Comment by zentol** [comment]
The test should be changed to use more specific folder names including a random component. If the directory we are trying to create already exists we should most definitely not just delete everything there, but instead try a different directory.

**Comment by shijinkui** [comment]
> The test should be changed to use more specific folder names including a random component. If the directory we are trying to create already exists we should not delete everything there (since we can't guarantee that we created those files), but instead try a different directory.

@zentol That's file. Also delete the tmp dir when exist.

File: flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java [hunk 1]
```diff
@@ -90,7 +91,9 @@ public void testSimpleRead() {
 
 	@Test
 	public void testNestedFileRead() {
-		String[] dirs = new String[] {"tmp/first/", "tmp/second/"};
+		String tmpDirName = "tmp-" + UUID.randomUUID().toString().substring(0, 8);
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java]: We could reuse `CommonTestUtils#createTempDirectory()'  in flink-runtime by copying the method into the CommonTestUtils class in `flink-test-utils-junit`.

File: flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java [hunk 2]
```diff
@@ -90,23 +92,23 @@ public void testSimpleRead() {
...
+			String[] dirs = new String[]{tmpDirPath + "/first/", tmpDirPath + "/second/"};
+			List<String> expectedFiles = new ArrayList<>();
+			for (String dir : dirs) {
 				// create input file
 				File tmpDir = new File(dir);
-				if (!tmpDir.exists()) {
-					tmpDir.mkdirs();
+				if (!tmpDir.exists() && tmpDir.mkdirs()) {
+					tmpDir.deleteOnExit();
 				}
```
Comment (zentol) [inline flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java]: this section could be a bit cleaner imo:
```
File parentDir = CommonTestUtils.createTempDirectory();
String[] dirs = new String[]{"first", "second"};
...
	File tmpDir= new File(parentDir , dir);
```

**Pull request closed without merging**